Vic2ToHoI4/Source/V2World/Localisations/LocalisationsFactory.cpp

```diff
@@ -0,0 +1,126 @@
+#include "LocalisationsFactory.h"
+#include "Configuration.h"
+#include "Log.h"
+#include "OSCompatibilityLayer.h"
+#include <array>
+#include <fstream>
+
+
+
+std::unique_ptr<Vic2::Localisations> Vic2::Localisations::Factory::importLocalisations(
+	 const Configuration& theConfiguration)
+{
+	Log(LogLevel::Info) << "Reading Vic2 localisation";
+
+	ReadFromAllFilesInFolder(theConfiguration.getVic2Path() + "/localisation");
+
+	for (const auto& mod: theConfiguration.getVic2Mods())
+	{
+		Log(LogLevel::Info) << "\tReading mod localisation";
+		ReadFromAllFilesInFolder(theConfiguration.getVic2ModPath() + "/" + mod.getDirectory() + "/localisation");
+	}
+
+	if (commonItems::DoesFileExist("DataFiles/Vic2Localisations.csv"))
+	{
+		ReadFromFile("DataFiles/Vic2Localisations.csv");
+	}
+
+	return std::make_unique<Localisations>(localisations, localisationToKeyMap);
+}
+
+
+void Vic2::Localisations::Factory::ReadFromAllFilesInFolder(const std::string& folderPath)
+{
+	for (const auto& fileName: commonItems::GetAllFilesInFolder(folderPath))
+	{
+		ReadFromFile(folderPath + '/' + fileName);
+	}
+}
+
+
+void Vic2::Localisations::Factory::ReadFromFile(const std::string& fileName)
+{
+	std::ifstream in(fileName);
+
+	while (!in.eof())
+	{
+		std::string line;
+		getline(in, line);
+		if (line[0] != '#')
+		{
+			processLine(line);
+		}
+	}
+
+	in.close();
+}
+
+
+constexpr std::array<std::pair<const char*, Vic2::Encoding>, 13> languages = {{
+	 {"english", Vic2::Encoding::Win1252},
+	 {"french", Vic2::Encoding::Win1252},
+	 {"german", Vic2::Encoding::Win1252},
+	 {"polish", Vic2::Encoding::Win1250},
+	 {"spanish", Vic2::Encoding::Win1252},
+	 {"italian", Vic2::Encoding::Win1252},
+	 {"swedish", Vic2::Encoding::Win1250},
+	 {"czech", Vic2::Encoding::Win1250},
+	 {"hungarian", Vic2::Encoding::Win1250},
+	 {"dutch", Vic2::Encoding::Win1252},
+	 {"braz_por", Vic2::Encoding::Win1252},
+	 {"russian", Vic2::Encoding::Win1251},
+	 {"finnish", Vic2::Encoding::Win1252},
+}};
+void Vic2::Localisations::Factory::processLine(const std::string& line)
+{
+	auto division = line.find_first_of(';');
+	const auto key = line.substr(0, division);
+
+	for (const auto& [language, encoding]: languages)
+	{
+		auto [rawLocalisation, newDivision] = extractNextLocalisation(line, division);
+		division = newDivision;
+
+		auto UTF8Result = convertToUtf8(rawLocalisation, encoding);
+
+		if (strcmp(language, "english") == 0)
```

Comment:
strcmp is C, not C++. 
if language == "english"...

Reply:
Yeah, I don't like this. But for complicated reasons explained in a non-public C++ tip, using string in the languages list is really quite bad. You get a good static data structure by making it constexpr, but that requires using char*. == can't be used, because it's now comparing against another char*. Thus we're into C-land.